buildSrc/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionBwcSetupPlugin.java

```diff
@@ -194,28 +206,36 @@ static void createBuildBwcTask(
         Provider<Version> bwcVersion,
         String projectName,
         String projectPath,
-        File projectArtifact,
-        TaskProvider<Task> bwcTaskProvider
+        DistributionProjectArtifact projectArtifact,
+        TaskProvider<Task> bwcTaskProvider,
+        String assembleTaskName
     ) {
         String bwcTaskName = buildBwcTaskName(projectName);
         bwcSetupExtension.bwcTask(bwcTaskName, c -> {
+            boolean useNativeExpanded = projectArtifact.expandedDistDir != null;
+            File expectedOutputFile = useNativeExpanded ? projectArtifact.expandedDistDir : projectArtifact.distFile;
             c.getInputs().file(new File(project.getBuildDir(), "refspec"));
-            c.getOutputs().files(projectArtifact);
+            if (useNativeExpanded) {
+                c.getOutputs().dir(expectedOutputFile);
+            } else {
+                c.getOutputs().files(expectedOutputFile);
+            }
             c.getOutputs().cacheIf("BWC distribution caching is disabled on 'master' branch", task -> {
                 String gitBranch = System.getenv("GIT_BRANCH");
                 return BuildParams.isCi() && (gitBranch == null || gitBranch.endsWith("master") == false);
             });
-            c.args(projectPath.replace('/', ':') + ":assemble");
+            c.args(projectPath.replace('/', ':') + ":" + assembleTaskName);
             if (project.getGradle().getStartParameter().isBuildCacheEnabled()) {
                 c.args("--build-cache");
             }
             c.doLast(task -> {
-                if (projectArtifact.exists() == false) {
+                if (expectedOutputFile.exists() == false) {
```

Comment:
One thing we lose here with the directory approach is the check that this created the correct version. When we forget to bump versions in branches this will error in a scenario where we expect it to produce an artifact with say, version 7.11.0 but instead 7.12.0 is created. This is an indicator a version bump was missed and we are not testing the thing we expect.

That works with the file approach, as the version is embedded in the expected file name, but here we are simply looking for a folder `.../build/install` at which point that could container an exploded install directory with any version name. The expected build artifact should take this into account. We should specifically ensure the exploded folder name matches the expected version.

Reply:
Might also be worth adding a test case for this.

Reply:
will fix this in a separate pr as it involves fixing it on the producer and consumer side (across branches)

Reply:
Let's make sure we do this sooner rather than later as this is one of the primary ways we are notified that version bumps got messed up.

Reply:
> and consumer side (across branches)

I dont' think that's true as the `install/` directory contains the exploded folder which looks something like `elasticsearch-8.0.0-snapshot/` so we just need only check that the install directory contains another folder with the name we expect.